themes/default/content/blog/organizing-aws-accounts-with-pulumi/index.md

```diff
@@ -0,0 +1,290 @@
+
+## Tagging Policies
+
+In keeping with our general theme here, we’ll encapsulate our tagging policy with yet another component resource so that we can easily include future org units. Similar to the backup policy syntax, the tag policy syntax uses JSON with some special placeholder variables.
+
+Let’s take a look at the consumer point-of-view for this component. We want users of this component to pass in a list of allowed cost centers for a certain org unit. Say, if a team wants to have multiple org units and therefore different cost centers for each of them, this model allows those teams to use the `TagPolicies` component for each org unit separately and assign the allowed cost center values accordingly.
+
+```typescript
+const tagPolicies = new TagPolicies("tagPolicies", {
+   costCenters: [
+       {
+           allowedCostCenters: ["Development", "Testing"],
+           ou: devOrgUnit,
+       },
+   ],
+   orgId: organization.then((o) => o.roots[0].id),
+});
+```
+
+The tag policy for our example looks like this:
+
+```typescript
+{
+   type: "TAG_POLICY",
+   content: JSON.stringify({
+       tags: {
+           CostCenter: {
+               tag_value: {
+                   "@@assign": costCenter.allowedCostCenters,
+               },
+           },
+
+           // Add tags that are enforceable only to the OU.
+           Owner: {
+               tag_key: {
+                   "@@assign": "Owner",
+                   // Don't allow any child policies from changing the key of this tag.
+                   "@@operators_allowed_for_child_policies": [
+                       "@@none",
+                   ],
+               },
+           },
+       },
+   }),
+}
+```
+
+And that’s about it as far as setting up the simple org unit and member account structure shown at the beginning of this post.
+View the [full source-code](https://github.com/pulumi/examples/tree/master/aws-ts-organizations) for the component resources in the `pulumi/examples` repository.
+
+This post demonstrated how you could use the Pulumi programming model to encapsulate and account for the hierarchical organization structure that your team desires or is required to have for any number of reasons. While an enterprise organization might be interested in this sort of a setup to ensure multiple teams in the organization operate in a similar way, smaller teams should also strongly consider organizing their AWS accounts in a way that enforces the principle of least privileged access. Especially, if you are looking to get SOC2 certified and need to ensure strict isolation of resources that hold any kind of real-world customer data.
+
+## Next Steps
```

Comment:
I'd really like to see what this looks like when it runs! Partly because as you suggest, many users won't have the appropriate permissions to actually deploy it. If there were some way to show what happens after a successful run, it'd really bring it all home I think. Up to you though.

Reply:
Yes definitely agree. Do you think a GIF or a screenshot of the Pulumi plan would be good here?

Reply:
An animated GIF that shows part of the `pulumi up` output would be cool, I think.

Reply:
I've expanded the outro a bit more along with a gif. Let me know what you think. If you would like to see just those changes, it would be commit 44254a42d6429d12975fe39ac3478610a80df619.